system/security/LdapSecurity/ldapconnection.cpp

```diff
@@ -1121,30 +1121,40 @@ class CLdapClient : public CInterface, implements ILdapClient
     virtual void init(IPermissionProcessor* pp)
     {
         m_pp = pp;
-        if(m_ldapconfig->getServerType() == OPEN_LDAP)
+        static CriticalSection  lcCrit;
```

Comment:
It is better if this static is defined outside the function.

If there is a static inside a function it is technically initialised when the function is first called, not when the program starts.  On modern compilers that means that there will need to be thread protection around the initialisation.